Gems/Atom/RHI/Vulkan/Code/Source/RHI/FrameGraphExecuteGroupHandlerBase.cpp

```diff
@@ -27,6 +27,12 @@ namespace AZ
         {
             EndInternal();
             m_device->GetCommandQueueContext().GetCommandQueue(m_hardwareQueueClass).ExecuteWork(AZStd::move(m_workRequest));
+#if defined(AZ_FORCE_CPU_GPU_INSYNC)
+            //Cache the name of the scope we just queued and wait for it to finish on the cpu
+            m_device->SetLastExecutingScope(m_workRequest.m_commandList->GetName().GetStringView());
+            m_device->GetCommandQueueContext().GetCommandQueue(m_hardwareQueueClass).FlushCommands();
+            m_device->GetCommandQueueContext().GetCommandQueue(m_hardwareQueueClass).WaitForIdle();
```

Comment:
nit: store `m_device->GetCommandQueueContext().GetCommandQueue(m_hardwareQueueClass)` to a local variable rather than call it 3 times.